crates/typst-library/src/model/outline.rs

```diff
@@ -1,49 +1,60 @@
-use crate::diag::{bail, At, SourceResult};
+use crate::diag::{bail, error, At, HintedStrResult, SourceResult, StrResult};
 use crate::engine::Engine;
 use crate::foundations::{
-    cast, elem, scope, select_where, Content, Context, Func, LocatableSelector,
-    NativeElement, Packed, Show, ShowSet, Smart, StyleChain, Styles,
+    cast, elem, func, scope, select_where, Args, Construct, Content, Context, Func,
+    LocatableSelector, NativeElement, Packed, Resolve, Show, ShowSet, Smart, StyleChain,
+    Styles,
 };
-use crate::introspection::{Counter, CounterKey, Locatable};
-use crate::layout::{
-    BoxElem, Dir, Em, Fr, HElem, HideElem, Length, Rel, RepeatElem, Spacing,
+use crate::introspection::{
+    Counter, CounterKey, Introspector, Locatable, Location, Locator, LocatorLink,
 };
-use crate::model::{
-    Destination, HeadingElem, NumberingPattern, ParElem, ParbreakElem, Refable,
+use crate::layout::{
+    Abs, Axes, BlockBody, BlockElem, BoxElem, Dir, Em, Fr, HElem, Length, Region, Rel,
+    RepeatElem, Sides,
 };
-use crate::text::{LinebreakElem, LocalName, SpaceElem, TextElem};
+use crate::math::EquationElem;
+use crate::model::{Destination, HeadingElem, NumberingPattern, ParElem, Refable};
+use crate::text::{LocalName, SpaceElem, TextElem};
 
 /// A table of contents, figures, or other elements.
 ///
 /// This function generates a list of all occurrences of an element in the
-/// document, up to a given depth. The element's numbering and page number will
-/// be displayed in the outline alongside its title or caption. By default this
-/// generates a table of contents.
+/// document, up to a given [`depth`]($outline.depth). The element's numbering
+/// and page number will be displayed in the outline alongside its title or
+/// caption.
 ///
 /// # Example
 /// ```example
+/// #set heading(numbering: "1.")
 /// #outline()
 ///
 /// = Introduction
 /// #lorem(5)
 ///
-/// = Prior work
+/// = Methods
+/// == Setup
 /// #lorem(10)
 /// ```
 ///
 /// # Alternative outlines
+/// In its default configuration, this function generates a table of contents.
 /// By setting the `target` parameter, the outline can be used to generate a
-/// list of other kinds of elements than headings. In the example below, we list
-/// all figures containing images by setting `target` to `{figure.where(kind:
-/// image)}`. We could have also set it to just `figure`, but then the list
-/// would also include figures containing tables or other material. For more
+/// list of other kinds of elements than headings.
+///
+/// In the example below, we list all figures containing images by setting
+/// `target` to `{figure.where(kind: image)}`. Just the same, we could have set
+/// it to `{figure.where(kind: table)}` to generate a list of tables.
+///
+/// We could also set it to just `figure`, but then the list would _all_
+/// figures, be it ones containing images, tables, or other material. For more
 /// details on the `where` selector, [see here]($function.where).
```

Comment:
```suggestion
/// We could also set it to just `figure`, without using a
/// [`where`]($function.where) selector, but then the list would contain _all_
/// figures, be it ones containing images, tables, or other material.
```

I believe it is generally bad practice for accessibility to have links with bodies such as "here," or "see here." The same link could be provided more organically using a different phrasing.

Regardless, the first sentence is missing a verb.